github-actions/open_pr.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 SUSE
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit -o nounset -o pipefail
+
+prs=$(
+  curl \
+    --fail \
+    --request GET \
+    --header "Accept: application/vnd.github.v3+json" \
+    "https://api.github.com/repos/${REPOSITORY}/pulls?state=open&head=${REPOSITORY_OWNER}:${BRANCH}" \
+    | jq -r length
+)
+
+if [ "${prs}" -gt 0 ]; then
+  >&2 echo "Pull request already exists for '${BRANCH}'"
+  exit 0
+fi
+
+body=$(jq -r -c . <<EOF
```

Comment:
`-c` is not necessary (because extra whitespace is harmless); while we _could_ do `body="{…}"` here too, the quoting probably means it's too much.

Reply:
Yes - but it's also used in machine-to-machine communication, so the `-c` is totally harmless. Nobody is expected to read the final transmission of the JSON string.